src/main/java/com/systemmeltdown/meltdownlog/topics/LogTopicRegistry.java

```diff
@@ -0,0 +1,60 @@
+package com.systemmeltdown.meltdownlog.topics;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Holds list of registered topics and allows sessions to subscribe to them.
+ */
+public class LogTopicRegistry {
+	private static LogTopicRegistry instance;
+
+	/**
+	 * Gets the current instance of the registry.
+	 */
+	public static LogTopicRegistry getInstance() {
+		if (instance == null) {
+			instance = new LogTopicRegistry();
+		}
+		return instance;
+	}
+
+	private final Map<String, LogTopic> m_topics;
+
+	/**
+	 * Creates new topic registry.
+	 * 
+	 * This constructor is intentionally not public to restrict creation of registries
+	 * to within this package.
+	 */
+	protected LogTopicRegistry() {
+		m_topics = new HashMap<String, LogTopic>();
+	}
+
+	/**
+	 * Gets a topic by its name.
+	 * 
+	 * @param topicName The name of the topic to get
+	 * @return The topic or null if not found
+	 */
+	public LogTopic getTopic(String topicName) {
+		return m_topics.get(topicName);
+	}
+
+	/**
+	 * Adds a topic to this registry.
+	 * 
+	 * Note: This function intentionally not public to restrict usage to this package.
```

Comment:
Shouldn't it be package-private instead of protected then? Since it's a singleton there shouldn't be subclasses.

Reply:
Changed to package (default) access. Thanks!